### PR TITLE
[MRG] Expose coord transform

### PR DESCRIPTION
**Pull request by KamalakerDadi**
I addressed remaining comments and took over the PR #1530 
Can I have final reviews ?

### CODE REVIEW

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -103,25 +104,43 @@ def coord_transform(x, y, z, affine):
...
+    return_number = isinstance(x, numbers.Number)
+    x = np.asanyarray(x)
+    shape = x.shape
     coords = np.c_[np.atleast_1d(x).flat,
                    np.atleast_1d(y).flat,
                    np.atleast_1d(z).flat,
                    np.ones_like(np.atleast_1d(z).flat)].T
     x, y, z, _ = np.dot(affine, coords)
+    if return_number:
+        return x.item(), y.item(), z.item()
```
Comment (bthirion) [inline nilearn/image/resampling.py]: Is there a unit test covering that case ?

**Comment by bthirion** [comment]
LGTM, besides the comment on testing all cases.

File: nilearn/image/resampling.py (hunk 1 above)
Comment (KamalakerDadi) [reply nilearn/image/resampling.py]: There is already in test_coord_transform module testing single integer as inputs and return output the single integer.

**Comment by KamalakerDadi** [comment]
Travis failure is because of compatibility issues with nibabel .get_affine and .affine

I removed get_affine because we plan to bump nibabel to 2.0.2 which I think should be fine after PR #1534 is merged.

**Comment by KamalakerDadi** [comment]
AppVeyor failure is unrelated and should be fixed with PR #1535

**Comment by bthirion** [comment]
LGTM then.

File: nilearn/image/resampling.py [hunk 2]
```diff
@@ -90,38 +91,56 @@ def coord_transform(x, y, z, affine):
...
+    Returns
+    -------
+    x : number or ndarray (same shape as input)
+        The x coordinates in the output space
+    y : number or ndarray (same shape as input)
+        The y coordinates in the output space
+    z : number or ndarray (same shape as input)
+        The z coordinates in the output space
+
+    Warning: The x, y and z have their Talairach ordering, not 3D
```
Comment (bthirion) [inline nilearn/image/resampling.py]: A bit weird to talk about Talairach here. maybe 'output space (e.g. MNI) coordinate ordering'

**Comment by GaelVaroquaux** [comment]
Can you rebase on master, so that hopefully the tests pass.

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge if tests pass

**Comment by KamalakerDadi** [comment]
>Can you rebase on master, so that hopefully the tests pass.

Rebased. Travis is passed and waiting for CircleCI to be passed.

**Comment by GaelVaroquaux** [comment]
Thank you very much! Merging!

**Pull request merged**